SwiftyStoreKit/InAppReceipt.swift

```diff
@@ -0,0 +1,239 @@
+        return NSBundle.mainBundle().appStoreReceiptURL
+    }
+
+    static var data: NSData? {
+        if let receiptDataURL = URL, data = NSData(contentsOfURL: receiptDataURL) {
+            return data
+        }
+        return nil
+    }
+
+    // The base64 encoded receipt data.
+    static var base64EncodedString: String? {
+        return data?.base64EncodedStringWithOptions([])
+    }
+
+    // https://developer.apple.com/library/ios/releasenotes/General/ValidateAppStoreReceipt/Chapters/ValidateRemotely.html
+
+   /**
+    *  - Parameter receiptVerifyURL: receipt verify url (default: Test)
+    *  - Parameter password: Only used for receipts that contain auto-renewable subscriptions. Your app’s shared secret (a hexadecimal string).
+    *  - Parameter session: the session used to make remote call.
+    *  - Parameter completion: handler for result
+    */
+    class func verify(
+        receiptVerifyURL url: ReceiptVerifyURL = .Test,
+        password autoRenewPassword: String? = nil,
+        session: NSURLSession = NSURLSession.sharedSession(),
+        completion:(result: SwiftyStoreKit.VerifyReceiptResultType) -> ()) {
+
+            // If no receipt is present, validation fails.
+            guard let base64EncodedString = self.base64EncodedString else {
+                completion(result: .Error(error: .NoReceiptData))
+                return
+            }
+
+            // Create request
+            let storeURL = NSURL(string: url.rawValue)! // safe (until no more)
+            let storeRequest = NSMutableURLRequest(URL: storeURL)
+            storeRequest.HTTPMethod = "POST"
+
+
+            let requestContents :NSMutableDictionary = [ "receipt-data" : base64EncodedString]
+            // password if defined
+            if let password = autoRenewPassword {
+                requestContents.setValue(password, forKey: "password")
+            }
+
+            // Encore request body
+            do {
+                storeRequest.HTTPBody = try NSJSONSerialization.dataWithJSONObject(requestContents, options: [])
+            } catch let e {
+                completion(result: .Error(error: .RequestBodyEncodeError(error: e)))
+                return
+            }
+
+            // Remote task
+            let task = session.dataTaskWithRequest(storeRequest) { data, response, error -> Void in
+
+                // there is an error
+                if let networkError = error {
+                    completion(result: .Error(error: .NetworkError(error: networkError)))
+                    return
+                }
+
+                // there is no data
+                guard let safeData = data else {
+                    completion(result:.Error(error: .NoRemoteData))
+                    return
+                }
+
+                // cannot decode data
+                guard let receiptInfo = try? NSJSONSerialization.JSONObjectWithData(data!, options: .MutableLeaves) as? ReceiptInfo ?? [:] else {
```

Comment:
can use ```safeData``` as already unwrapped here:
```
guard let receiptInfo = try? NSJSONSerialization.JSONObjectWithData(safeData, options: .MutableLeaves) as? ReceiptInfo else
```
Also should the ```?? [:]``` at the end be removed so that we can report JSONDecodeError if we can't get a non-nil ```ReceiptInfo```?